### ISSUE TITLE
[0.33.7.1] Cannot create new users or send emails (pulses/alerts/invitations)

### ISSUE THREAD

**Issue by flamber** [metabase/metabase#11525]
**Describe the bug**
Cannot create new users on 0.33.7.1, or sending pulse mails.

**Logs**
When trying to create a user:
```
12-15 11:43:04 ERROR middleware.log :: POST /api/user 500 247.0 ms (9 DB calls) 
{:message "No implementation of method: :render of protocol: #'stencil.ast/ASTNode found for class: nil",
 :type java.lang.IllegalArgumentException,
 :stacktrace
 ("clojure.core$_cache_protocol_fn.invokeStatic(core_deftype.clj:583)"
  "clojure.core$_cache_protocol_fn.invoke(core_deftype.clj:575)"
  "stencil.ast$fn__30303$G__30298__30312.invoke(ast.clj:19)"
  "stencil.core$render.invokeStatic(core.clj:80)"
  "stencil.core$render.invoke(core.clj:74)"
  "stencil.core$render_file.invokeStatic(core.clj:87)"
  "stencil.core$render_file.invoke(core.clj:83)"
  "--> email.messages$send_new_user_email_BANG_.invokeStatic(messages.clj:67)"
  "email.messages$send_new_user_email_BANG_.invoke(messages.clj:63)"
  "models.user$send_welcome_email_BANG_.invokeStatic(user.clj:176)"
  "models.user$send_welcome_email_BANG_.invoke(user.clj:172)"
  "models.user$fn__32292$create_and_invite_user_BANG___32297$fn__32298.invoke(user.clj:208)"
  "models.user$fn__32292$create_and_invite_user_BANG___32297.invoke(user.clj:203)"
  "api.user$fn__63972$fn__63975.invoke(user.clj:101)"
  "api.user$fn__63972.invokeStatic(user.clj:100)"
  "api.user$fn__63972.invoke(user.clj:89)"
  "middleware.auth$enforce_authentication$fn__64360.invoke(auth.clj:14)"
  "routes$fn__65737$fn__65738.doInvoke(routes.clj:56)"
  "middleware.exceptions$catch_uncaught_exceptions$fn__64459.invoke(exceptions.clj:104)"
  "middleware.exceptions$catch_api_exceptions$fn__64456.invoke(exceptions.clj:92)"
  "middleware.log$log_api_call$fn__66111$fn__66112.invoke(log.clj:170)"
  "middleware.log$log_api_call$fn__66111.invoke(log.clj:164)"
  "middleware.security$add_security_headers$fn__64422.invoke(security.clj:122)"
  "middleware.json$wrap_json_body$fn__65816.invoke(json.clj:61)"
  "middleware.json$wrap_streamed_json_response$fn__65834.invoke(json.clj:97)"
  "middleware.session$bind_current_user$fn__60984$fn__60985.invoke(session.clj:209)"
  "middleware.session$do_with_current_user.invokeStatic(session.clj:184)"
  "middleware.session$do_with_current_user.invoke(session.clj:177)"
  "middleware.session$bind_current_user$fn__60984.invoke(session.clj:208)"
  "middleware.session$wrap_current_user_id$fn__60969.invoke(session.clj:161)"
  "middleware.session$wrap_session_id$fn__60954.invoke(session.clj:123)"
  "middleware.auth$wrap_api_key$fn__64368.invoke(auth.clj:27)"
  "middleware.misc$maybe_set_site_url$fn__66138.invoke(misc.clj:56)"
  "middleware.misc$bind_user_locale$fn__66141.invoke(misc.clj:72)"
  "middleware.misc$add_content_type$fn__66126.invoke(misc.clj:28)"
  "middleware.misc$disable_streaming_buffering$fn__66149.invoke(misc.clj:87)")}
```
When trying to send pulse test email:
```
12-15 12:49:41 ERROR middleware.log :: POST /api/pulse/test 500 187.6 ms (15 DB calls) 
{:message "No implementation of method: :render of protocol: #'stencil.ast/ASTNode found for class: nil",
 :type java.lang.IllegalArgumentException,
 :stacktrace
...
```
When sending scheduled pulse:
```
12-15 13:00:00 INFO task.send-pulses :: Sending scheduled pulses...
12-15 13:00:00 ERROR task.send-pulses :: Error sending Pulse 1
java.lang.IllegalArgumentException: No implementation of method: :render of protocol: #'stencil.ast/ASTNode found for class: nil
...
```

**To Reproduce**
- Go to Admin > People > Add someone > fill in details and click Create
- Go to a pulse and click Send test email
- Go to a pulse and schedule a send, and check the log when pulse should have been sent
- ...perhaps there's other sections not working - will update as I find them...

**Information about your Metabase Installation:**
Metabase 0.33.7.1 Docker and JAR on various hosting

**Severity** P0

**Additional context**
https://discourse.metabase.com/t/cant-add-a-new-user-no-implementation-of-method-render-error/8350

**Comment by leeuwd** [comment]
Same here when inviting user: `No implementation of method: :render of protocol: #'stencil.ast/ASTNode found for class: nil`.

**Comment by philippelawson** [comment]
Same here when trying to reset user password:
No implementation of method: :render of protocol: #'stencil.ast/ASTNode found for class: nil

**Comment by mazameli** [comment]
Just adding a note that the user creation error occurs even if you have not yet set up SMTP or auth. I'm getting `No implementation of method: :render of protocol: #'stencil.ast/ASTNode found for class: nil`

![image](https://user-images.githubusercontent.com/2223916/70931942-6b87e600-1fed-11ea-9d46-ad3cdecbae6f.png)

Server log:

```
12-16 10:17:08 ERROR middleware.log :: POST /api/user 500 177.7 ms (7 DB calls) 
{:message "No implementation of method: :render of protocol: #'stencil.ast/ASTNode found for class: nil",
 :type java.lang.IllegalArgumentException,
 :stacktrace
 ("clojure.core$_cache_protocol_fn.invokeStatic(core_deftype.clj:583)"
  "clojure.core$_cache_protocol_fn.invoke(core_deftype.clj:575)"
  "stencil.ast$fn__30303$G__30298__30312.invoke(ast.clj:19)"
  "stencil.core$render.invokeStatic(core.clj:80)"
  "stencil.core$render.invoke(core.clj:74)"
  "stencil.core$render_file.invokeStatic(core.clj:87)"
  "stencil.core$render_file.invoke(core.clj:83)"
  "--> email.messages$send_new_user_email_BANG_.invokeStatic(messages.clj:67)"
  "email.messages$send_new_user_email_BANG_.invoke(messages.clj:63)"
  "models.user$send_welcome_email_BANG_.invokeStatic(user.clj:176)"
  "models.user$send_welcome_email_BANG_.invoke(user.clj:172)"
  "models.user$fn__32292$create_and_invite_user_BANG___32297$fn__32298.invoke(user.clj:208)"
  "models.user$fn__32292$create_and_invite_user_BANG___32297.invoke(user.clj:203)"
  "api.user$fn__63972$fn__63975.invoke(user.clj:101)"
  "api.user$fn__63972.invokeStatic(user.clj:100)"
  "api.user$fn__63972.invoke(user.clj:89)"
  "middleware.auth$enforce_authentication$fn__64360.invoke(auth.clj:14)"
  "routes$fn__65737$fn__65738.doInvoke(routes.clj:56)"
  "middleware.exceptions$catch_uncaught_exceptions$fn__64459.invoke(exceptions.clj:104)"
  "middleware.exceptions$catch_api_exceptions$fn__64456.invoke(exceptions.clj:92)"
  "middleware.log$log_api_call$fn__66111$fn__66112.invoke(log.clj:170)"
  "middleware.log$log_api_call$fn__66111.invoke(log.clj:164)"
  "middleware.security$add_security_headers$fn__64422.invoke(security.clj:122)"
  "middleware.json$wrap_json_body$fn__65816.invoke(json.clj:61)"
  "middleware.json$wrap_streamed_json_response$fn__65834.invoke(json.clj:97)"
  "middleware.session$bind_current_user$fn__60984$fn__60985.invoke(session.clj:209)"
  "middleware.session$do_with_current_user.invokeStatic(session.clj:184)"
  "middleware.session$do_with_current_user.invoke(session.clj:177)"
  "middleware.session$bind_current_user$fn__60984.invoke(session.clj:208)"
  "middleware.session$wrap_current_user_id$fn__60969.invoke(session.clj:161)"
  "middleware.session$wrap_session_id$fn__60954.invoke(session.clj:123)"
  "middleware.auth$wrap_api_key$fn__64368.invoke(auth.clj:27)"
  "middleware.misc$maybe_set_site_url$fn__66138.invoke(misc.clj:56)"
  "middleware.misc$bind_user_locale$fn__66141.invoke(misc.clj:72)"
  "middleware.misc$add_content_type$fn__66126.invoke(misc.clj:28)"
  "middleware.misc$disable_streaming_buffering$fn__66149.invoke(misc.clj:87)")}
```

**Comment by mazameli** [comment]
Another note: SMTP test emails seem to send fine.

Sending a test Pulse results in a sending failed error with this in the logs, so it appears the problem in both the pulse and auth cases has to do with the [Stencil](https://cljdoc.org/d/stencil/stencil/0.5.0/api/stencil.ast) dep.

```
12-16 10:27:29 ERROR middleware.log :: POST /api/pulse/test 500 125.9 ms (15 DB calls) 
{:message "No implementation of method: :render of protocol: #'stencil.ast/ASTNode found for class: nil",
 :type java.lang.IllegalArgumentException,
 :stacktrace
 ("clojure.core$_cache_protocol_fn.invokeStatic(core_deftype.clj:583)"
  "clojure.core$_cache_protocol_fn.invoke(core_deftype.clj:575)"
  "stencil.ast$fn__30303$G__30298__30312.invoke(ast.clj:19)"
  "stencil.core$render.invokeStatic(core.clj:80)"
  "stencil.core$render.invoke(core.clj:74)"
  "stencil.core$render_file.invokeStatic(core.clj:87)"
  "stencil.core$render_file.invoke(core.clj:83)"
  "--> email.messages$render_message_body.invokeStatic(messages.clj:267)"
  "email.messages$render_message_body.invoke(messages.clj:261)"
  "email.messages$render_pulse_email.invokeStatic(messages.clj:279)"
  "email.messages$render_pulse_email.invoke(messages.clj:276)"
  "pulse$fn__61109.invokeStatic(pulse.clj:169)"
  "pulse$fn__61109.invoke(pulse.clj:160)"
  "pulse$results__GT_notifications$iter__61139__61143$fn__61144.invoke(pulse.clj:212)"
  "pulse$send_notifications_BANG_.invokeStatic(pulse.clj:248)"
  "pulse$send_notifications_BANG_.invoke(pulse.clj:247)"
  "pulse$send_pulse_BANG_.invokeStatic(pulse.clj:274)"
  "pulse$send_pulse_BANG_.doInvoke(pulse.clj:256)"
  "api.pulse$fn__61672.invokeStatic(pulse.clj:195)"
  "api.pulse$fn__61672.invoke(pulse.clj:185)"
  "middleware.auth$enforce_authentication$fn__64360.invoke(auth.clj:14)"
  "routes$fn__65737$fn__65738.doInvoke(routes.clj:56)"
  "middleware.exceptions$catch_uncaught_exceptions$fn__64459.invoke(exceptions.clj:104)"
  "middleware.exceptions$catch_api_exceptions$fn__64456.invoke(exceptions.clj:92)"
  "middleware.log$log_api_call$fn__66111$fn__66112.invoke(log.clj:170)"
  "middleware.log$log_api_call$fn__66111.invoke(log.clj:164)"
  "middleware.security$add_security_headers$fn__64422.invoke(security.clj:122)"
  "middleware.json$wrap_json_body$fn__65816.invoke(json.clj:61)"
  "middleware.json$wrap_streamed_json_response$fn__65834.invoke(json.clj:97)"
  "middleware.session$bind_current_user$fn__60984$fn__60985.invoke(session.clj:209)"
  "middleware.session$do_with_current_user.invokeStatic(session.clj:184)"
  "middleware.session$do_with_current_user.invoke(session.clj:177)"
  "middleware.session$bind_current_user$fn__60984.invoke(session.clj:208)"
  "middleware.session$wrap_current_user_id$fn__60969.invoke(session.clj:161)"
  "middleware.session$wrap_session_id$fn__60954.invoke(session.clj:123)"
  "middleware.auth$wrap_api_key$fn__64368.invoke(auth.clj:27)"
  "middleware.misc$maybe_set_site_url$fn__66138.invoke(misc.clj:56)"
  "middleware.misc$bind_user_locale$fn__66141.invoke(misc.clj:72)"
  "middleware.misc$add_content_type$fn__66126.invoke(misc.clj:28)"
  "middleware.misc$disable_streaming_buffering$fn__66149.invoke(misc.clj:87)")}
```

**Comment by camsaul** [comment]
Looks like this error is happening because something getting passed into the Mustache template is `nil` (Stencil is the Mustache lib we use.) So the right place to start looking for bugs is with what gets passed to the templates. Whether anything has changed recently. Settings (e.g. email title?) Or maybe the email templates changed?

**Comment by mazameli** [comment]
@flyingmachine and @camsaul, FYI, I just tried reproducing this on the corresponding Enterprise Edition version, 1.33.7,  but I couldn't. User creation, both with and without SMTP set up, as well as pulses both seem to work fine in 1.33.7.

**Comment by flamber** [comment]
This issue has been fixed in 0.33.7.3, which was released a few minutes ago. Please upgrade :+1:
